source/session.ts

```diff
@@ -623,7 +527,7 @@ export class Session {
       additionalHeaders = {},
       decodeDatesAsIso = false,
     }: CallOptions = {}
-  ): Promise<T[]> {
+  ): Promise<IsTuple<T> extends true ? T : T[]> {
```

Comment:
add @typeParam for how this generic works https://tsdoc.org/pages/tags/typeparam/